### PR TITLE
Patches 2 upstream systemd

### PR DESCRIPTION
**Pull request by rajmohanrgithub**
Message from commits is given below:

systemd: Add more logs for boot procedure analysis
systemd-analyze: Add svg scaling options
systemd: set suid_dumpable to "suidssafe"
mount: Remove files in place of mount points

### CODE REVIEW

**Comment by keszybz** [comment]
Please set the title of the pull request.

File: src/core/main.c [hunk 1]
```diff
@@ -1699,6 +1699,17 @@ static void cmdline_take_random_seed(void) {
                    "This functionality should not be used outside of testing environments.");
 }
 
+static void set_suid_dumpable(void) {
+        const char *sd_path = "/proc/sys/fs/suid_dumpable";
```
Comment (mrc0mmand) [inline src/core/main.c]: Uh, we already do this? https://github.com/systemd/systemd/blob/1b47cfab7f34c7c4a49d3320080398c206bc7f82/sysctl.d/50-coredump.conf.in#L37

File: src/core/main.c [hunk 2]
```diff
@@ -2032,6 +2032,14 @@ static int invoke_main_loop(
         assert(ret_switch_root_init);
         assert(ret_error_message);
 
+        log_warning("sd: i=userspace, inactive_exit="USEC_FMT, m->timestamps[MANAGER_TIMESTAMP_USERSPACE].monotonic);
```
Comment (mrc0mmand) [inline src/core/main.c]: If anything, these definitely shouldn't be logged at the warning level (debug seems to be appropriate).

File: src/core/mount.c [hunk 3]
```diff
@@ -1168,6 +1169,16 @@ static void mount_enter_mounting(Mount *m) {
         m->control_command_id = MOUNT_EXEC_MOUNT;
         m->control_command = m->exec_command + MOUNT_EXEC_MOUNT;
 
+        dp = opendir(m->where);
```
Comment (mrc0mmand) [inline src/core/mount.c]: I don't think we want to nuke any files from the filesystem which are not under our direct control.

**Comment by mrc0mmand** [comment]
Also, please follow https://github.com/systemd/systemd/blob/main/docs/CODING_STYLE.md.

File: src/core/main.c (hunk 2 above)
Comment (bluca) [reply src/core/main.c]: and they can all be queried at runtime, and some are already logged later, so doesn't seem worth adding any of these

File: src/analyze/analyze.c [hunk 4]
```diff
@@ -274,6 +276,8 @@ static int help(int argc, char *argv[], void *userdata) {
                "                             security review of the unit(s)\n"
                "     --unit=UNIT             Evaluate conditions and asserts of unit\n"
                "     --table                 Output plot's raw time data as a table\n"
+               "  -s --scale=FACTOR          Stretch the x-axis of the plot by FACTOR (default: 1.0)\n"
```
Comment (bluca) [inline src/analyze/analyze.c]: shorthands are prime real estate, and sd-analyze has tons of options and will get more. This is not something common enough to warrant taking over one of the 52 available single-letter options, so please just use the long one

**Comment by bluca** [comment]
Please split the sd-analyze commit into a separate PR, as that looks reasonable and could be merged quickly once fixed up

**Comment by YHNdnzj** [comment]
Sorry, but as mentioned, all commits except for the sd-analyze one make no sense at all. The timestamps can be queried through dbus, and we do not do the intense logging about them. The mount one is even more problematic. What's the rationale???

**Pull request closed without merging**